src/topology.rs

```diff
@@ -385,15 +383,17 @@ impl Topology {
     /// topology.add_bond(2, 3);
     /// assert_eq!(topology.bonds(), vec![[0, 1], [1, 2], [2, 3]]);
     /// ```
-    pub fn bonds(&self) -> Vec<[u64; 2]> {
-        let count = self.bonds_count();
-        #[allow(clippy::cast_possible_truncation)]
-        let size = count as usize;
+    pub fn bonds(&self) -> Vec<[usize; 2]> {
+        let size = self.impropers_count();
+        let count = size as u64;
         let mut bonds = vec![[u64::max_value(); 2]; size];
         unsafe {
             check_success(chfl_topology_bonds(self.as_ptr(), bonds.as_mut_ptr(), count));
         }
-        return bonds;
+        return bonds
+            .iter()
+            .map(|tuple| [tuple[0] as usize, tuple[1] as usize])
+            .collect();
```

Comment:
Not sure how to handle this properly, this feels unidiomatic. Any suggestions?

Reply:
Yeah, I think this is why the API had mixed usize/u64 in the first place: we need to allocate new memory for bonds/angles/... to transform them from u64 to usize. This is a bit unfortunate, and I don't know how to improve it here.

----

But the way you are transforming from Vec<[u64; 2]> to Vec<[usize; 2]> looks good to me. It might be possible to optimize it by using Vec::reserve instead, but this iterator-based code should already be fast enough.

Reply:
I would avoid naming the variable in map `tuple`, since it is not a tuple in Rust sense. How about `bond`?

Reply:
Renaming makes sense, done.
I just learned, that `collect`over iterators that are sized or owned tends to be [quite performant](https://doc.rust-lang.org/src/alloc/vec.rs.html#1974-2000). So I guess we wouldn't improve with `Vec::reserve`.